### DISCUSSION TITLE
Schema does not validate media-type special characters

### DISCUSSION THREAD

**Discussion by telosBA** [usnistgov/OSCAL#1253] [Q&A] [answered]
# Describe the bug
Schema does not validate media-type special characters e.g. “image/png” - doesn’t meet URI requirements.
{A clear and concise description of what the bug is.}

# Who is the bug affecting?
Telos
# What is affected by this bug?
Schema validation of XML files.
{Describe the impact the bug is having.}

# When does this occur?
When validating XML files that contain media-type elements with special characters.
{Describe the conditions under which the bug is occurring.}

# How do we replicate the issue?
Attempt to validate an XML file containing a media-type with a special character. e.g. “image/png”
{What are the steps to reproduce the behavior?
1. Do this...
1. Then this...
1. See error

If applicable, add screenshots to help explain your problem.}

# Expected behavior (i.e. solution)
Error on associated line.
{A clear and concise description of what you expected to happen.}

# Other Comments

{Add any other context about the problem here.}

**Comment by aj-stein-nist** [comment]
Can you please provide an example that evidences that such a `@media-type` does not pass validation? We checked and this data is of type string, so the `/` character should be acceptable and validate. Thanks!

**Comment by telosBA** [comment]
Here is the error output:

TCL-07068B:~ 13503$ xmllint --schema /Users/13503/PycharmProjects/OSCAL/oscal.xsd /Users/13503/PycharmProjects/OSCAL/NewXacta.xml --noout
/Users/13503/PycharmProjects/OSCAL/NewXacta.xml:459: element link: Schemas validity error : Element '{http://csrc.nist.gov/ns/oscal/1.0}link', attribute 'rel': [facet 'pattern'] The value 'image/png' is not accepted by the pattern '[\i-[:𐀀-󯿿]][\c-[:𐀀-󯿿]]*'.
/Users/13503/PycharmProjects/OSCAL/NewXacta.xml:459: element link: Schemas validity error : Element '{http://csrc.nist.gov/ns/oscal/1.0}link', attribute 'rel': 'image/png' is not a valid value of the atomic type '{http://csrc.nist.gov/ns/oscal/1.0}token'.
/Users/13503/PycharmProjects/OSCAL/NewXacta.xml fails to validate
TCL-07068B:~ 13503$

Would you also be able to clarify requirements/expectations for "rel" in this case?

**Comment by david-waltermire** [comment, marked as answer]
The media type does not get defined in the [`rel`](https://pages.nist.gov/OSCAL/reference/latest/complete/xml-definitions/#/assembly/oscal-metadata/link/rel). The [`media-type`](https://pages.nist.gov/OSCAL/reference/latest/complete/xml-definitions/#/assembly/oscal-metadata/link/media-type) field is the correct one for that information. Instead, the `rel` is used to define the relation type, which declares the nature of the linkage.

To clarify which `rel` value to use I would need to understand the purpose of the link.

**Reply by telosBA** [reply]
Noted, we misinterpreted the usage of `rel` This cleared up the issue.
